### DISCUSSION TITLE
Getting console error "fetch is not defined" and the page wont redirect

### DISCUSSION THREAD

**Discussion by nishant-shah-social** [cypress-io/cypress#18412] [Questions and Help]
While doing cypress execution, after entering username and password, when I click on the login button, it gives this error on console := "fetch is not defined..." and i wont be redirected to the next page...
However, when I perform the same steps manually on the same browser version, I dont get this error in console and things work fine...
Any idea? 
While browsing I came across this issue :- https://github.com/cypress-io/cypress/issues/687, which has mentions of how to intercept and stub Fetch requests...
But, my requirement here is not to intercept any request or stub it, its a normal flow which is not working for me...
i am using cypress 8.4 and chrome 94
Any help will be appreciated

**Comment by portyu9** [comment]
Hi @nishant-shah-social 

Chrome 94 already supports `fetch`, so Cypress should not normally make it undefined.

Check your Cypress support/setup code for an old workaround such as:

```js
cy.on('window:before:load', (win) => {
  delete win.fetch
})
```

or anything else overwriting `window.fetch`.

Remove that code and migrate any old `cy.server()` / `cy.route()` network stubbing to `cy.intercept()`.

You can confirm the issue immediately with:

```js
cy.window().its('fetch').should('be.a', 'function')
```

If that fails, something in your Cypress/test setup is removing `fetch`; the login redirect itself is only the downstream symptom.

Please mark this answer as accepted if it helped.

**Reply by portyu9** [reply]
@nishant-shah-social Please mark this answer as accepted if it helped, thank you.
